pkg/operator/update/resources.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2020 PlanetScale Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+)
+
+func ShardDiskSize(dst []planetscalev2.VitessShardTabletPool, src []planetscalev2.VitessShardTabletPool) {
+	for i := range dst {
+		srcTabletDiskSize := src[i].DataVolumeClaimTemplate.Resources.Requests[corev1.ResourceStorage]
+		dst[i].DataVolumeClaimTemplate.Resources.Requests[corev1.ResourceStorage] = srcTabletDiskSize
+	}
+}
+
+// KeyspaceDiskSize updates values in 'dst' based on values in 'src'.
+// It leaves extra entries (found in 'dst' but not in 'src') untouched,
+// since those might be set by mutating admission webhooks or other controllers.
+// TODO: Add unit test to check that various buried disk sizes get updated and
+// check that invalid cases (not matching) are left untouched. We can also check
+// that nothing besides the disk sizes is touched.
+func KeyspaceDiskSize(dst, src *planetscalev2.VitessKeyspaceTemplate) {
+	// Check that the keyspace definitions line up.
+	if !validateKeyspacePartitionings(dst.Partitionings, src.Partitionings) {
+		return
+	}
+
+	updateDiskSize(dst, src)
+}
+
+func updateDiskSize(dst, src *planetscalev2.VitessKeyspaceTemplate) {
+	for i := range dst.Partitionings {
+		dstPartitioning := &dst.Partitionings[i]
+		if dstPartitioning.Equal != nil {
+			updateEqualPartitioningDiskSize(dstPartitioning.Equal, src.Partitionings[i].Equal)
+		} else {
```

Comment:
Just to be as liberal as we can in what we accept:

```suggestion
		}
		if dstPartitioning.Custom != nil {
```